pkg/operator/ceph/cluster/cluster_cleanup.go

```diff
@@ -0,0 +1,185 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mgr"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/rbd"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/ceph/file/mds"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	clusterCleanUpPolicyRetryInterval = 5 //seconds
+)
+
+var (
+	volumnName      = "cleanup-volume"
+	appName         = "rook-ceph-cleanup"
+	dataDirHostPath = "ROOK_DATA_DIR_HOST_PATH"
+)
+
+func (c *ClusterController) startClusterCleanUp(cluster *cephv1.CephCluster, cephHosts map[string][]v1.Pod) {
+	logger.Infof("starting clean up for cluster %q", cluster.Name)
+	if len(cephHosts) == 0 {
+		logger.Errorf("failed to clean cluster %q as no running ceph daemons are found", cluster.Namespace)
+		return
+	}
+
+	err := c.waitForCephDaemonCleanUp(cluster, time.Duration(clusterCleanUpPolicyRetryInterval)*time.Second)
+	if err != nil {
+		logger.Errorf("failed to wait till ceph daemons are destroyed. %v", err)
+		return
+	}
+
+	c.startCleanUpJob(cluster, cephHosts)
+}
+
+func (c *ClusterController) startCleanUpJob(cluster *cephv1.CephCluster, cephHosts map[string][]v1.Pod) {
+	for node := range cephHosts {
+		logger.Infof("starting clean up job on node %q", node)
+		jobName := k8sutil.TruncateNodeName("cluster-cleanup-job-%s", node)
+		podSpec := c.cleanUpJobTemplateSpec(cluster)
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: node}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      jobName,
+				Namespace: cluster.Namespace,
+				Labels:    controller.AppLabels(appName, cluster.Namespace),
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		if err := k8sutil.RunReplaceableJob(c.context.Clientset, job, true); err != nil {
+			logger.Error("failed to run cluster clean up job on node %q. %+v", node, err)
+		}
+	}
+}
+
+func (c *ClusterController) cleanUpJobContainer(cluster *cephv1.CephCluster) v1.Container {
+	privileged := true
```

Comment:
Use `privilegedContext()` from the OSD code and export it, no need to replicate that behavior again :).